From 3146752fdb2b0d8a0f6d9aa36c105fe7de2eb43b Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 13 Mar 2023 15:24:08 +0200
Subject: [PATCH] [SW-116644] gaudi3: use new specs_h9 for register access

In order to reduce the specs header file sizes CAD team
came with new repo specs_h9. This patch adds support to
use register addresses from specs_h9 repo which uses block
base instead of config bar base. This aligns firmware
implementation with LKD.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: If4cf884cf58ac8ec3cda961dfa5e6db44125dfa1
---

diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index 2123bd6..d1ac6a3 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -3,11 +3,10 @@
  * (C) Copyright 2022 HabanaLabs, Ltd.
  */
 #include <gaudi3_blocks_linux_driver.h>
-#include <d0_psoc_global_conf_regs.h>
-#include <d0_psoc_global_conf_masks.h>
-#include <d0_parc_arc0_cfg_regs.h>
-#include <d0_parc_arc2_cfg_regs.h>
-#include <d0_parc_local_cfg_regs.h>
+#include <gaudi3_reg_map.h>
+#include <global_conf_regs.h>
+#include <global_conf_masks.h>
+#include <arc_regs.h>
 #include <common/pci_ids.h>
 #include <stddef.h>
 #include <bitops.h>
@@ -86,8 +85,8 @@
 
 void hl_uart_init(void)
 {
-	regmap_writel(1, mmD0_PSOC_GLOBAL_CONF_PAD_SEL_48);	/* UART0 RXD */
-	regmap_writel(1, mmD0_PSOC_GLOBAL_CONF_PAD_SEL_49);	/* UART0 TXD */
+	regmap_writel(1, D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_SEL_48)); /* UART0 RXD */
+	regmap_writel(1, D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_SEL_49)); /* UART0 TXD */
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
 
@@ -117,13 +116,14 @@
 	if (rst_src != HL_COLD_RST)
 		return;
 
-	regmap_writel(D0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL_START_M,
-		      mmD0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL);
+	regmap_writel(GLOBAL_CONF_SCRATCHPAD_INIT_CTRL_START_M,
+		      D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_SCRATCHPAD_INIT_CTRL));
 	start = timestamp_get_us();
 
 	while (1) {
-		init_sts = regmap_readl(mmD0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL);
-		if (init_sts & D0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL_DONE_M)
+		init_sts =
+			regmap_readl(D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_SCRATCHPAD_INIT_CTRL));
+		if (init_sts & GLOBAL_CONF_SCRATCHPAD_INIT_CTRL_DONE_M)
 			break;
 
 		/*
@@ -188,8 +188,8 @@
 void btldr_parc_prepare(void)
 {
 	/* ARC must not be in reset state when accessing the CCM modules */
-	regmap_writel(0, mmD0_PARC_ARC0_CFG_ARC_RST);
-	regmap_writel(0, mmD0_PARC_ARC2_CFG_ARC_RST);
+	regmap_writel(0, D0_PARC_ARC0_CFG_REG(mmARC_ARC_RST));
+	regmap_writel(0, mmD0_PARC_ARC2_CFG_BASE + mmARC_ARC_RST);
 }
 
 #define EFUSE_APP_SIZE_OFFSET	55
